packages/angular_devkit/build_angular/src/utils/service-worker.ts

```diff
@@ -5,67 +5,57 @@
-  relative,
-  tags,
-  virtualFs,
-} from '@angular-devkit/core';
-import { NodeJsSyncHost } from '@angular-devkit/core/node';
-import {
-  Filesystem,
-  Generator,
-} from '@angular/service-worker/config';
+import { Path, getSystemPath, normalize } from '@angular-devkit/core';
+import { Config, Filesystem, Generator } from '@angular/service-worker/config';
 import * as crypto from 'crypto';
+import { constants as fsConstants, createReadStream, promises as fs } from 'fs';
+import * as path from 'path';
+import { pipeline } from 'stream';
 
 class CliFilesystem implements Filesystem {
-  constructor(private _host: virtualFs.Host, private base: string) { }
+  constructor(private base: string) {}
 
-  list(path: string): Promise<string[]> {
-    return this._recursiveList(this._resolve(path), []).catch(() => []);
+  list(dir: string): Promise<string[]> {
+    return this._recursiveList(this._resolve(dir), []);
   }
 
-  async read(path: string): Promise<string> {
-    return virtualFs.fileBufferToString(await this._readIntoBuffer(path));
+  read(file: string): Promise<string> {
+    return fs.readFile(this._resolve(file), 'utf-8');
   }
 
-  async hash(path: string): Promise<string> {
-    const sha1 = crypto.createHash('sha1');
-    sha1.update(Buffer.from(await this._readIntoBuffer(path)));
-
-    return sha1.digest('hex');
-  }
-
-  write(path: string, content: string): Promise<void> {
-    return this._host.write(this._resolve(path), virtualFs.stringToFileBuffer(content))
-      .toPromise();
+  hash(file: string): Promise<string> {
+    return new Promise((resolve, reject) => {
+      const hash = crypto.createHash('sha1').setEncoding('hex');
+      pipeline(
+        createReadStream(this._resolve(file)),
+        hash,
+        (error) => error ? reject(error) : resolve(hash.read()),
+      );
+    });
   }
 
-  private _readIntoBuffer(path: string): Promise<ArrayBuffer> {
-    return this._host.read(this._resolve(path)).toPromise();
+  write(file: string, content: string): Promise<void> {
+    return fs.writeFile(this._resolve(file), content);
   }
 
-  private _resolve(path: string): Path {
-    return join(normalize(this.base), normalize(path));
+  private _resolve(file: string): string {
+    return path.join(this.base, file);
   }
 
-  private async _recursiveList(path: Path, items: string[]): Promise<string[]> {
-    const fragments = await this._host.list(path).toPromise();
-
-    for (const fragment of fragments) {
-      const item = join(path, fragment);
-
-      if (await this._host.isDirectory(item).toPromise()) {
-        await this._recursiveList(item, items);
-      } else {
-        items.push('/' + relative(normalize(this.base), item));
+  private async _recursiveList(dir: string, items: string[]): Promise<string[]> {
+    const subdirectories = [];
+    for await (const entry of await fs.opendir(dir)) {
+      if (entry.isFile()) {
+        items.push('/' + path.relative(this.base, path.join(dir, entry.name)));
```

Comment:
Does the SW generator handle non posix path?

Reply:
apparently not. updated to use posix path variants.